packages/services/api/src/modules/audit-logs/providers/audit-logs-manager.ts

```diff
@@ -0,0 +1,152 @@
+import { Injectable, Scope } from 'graphql-modules';
+import { z } from 'zod';
+import * as Sentry from '@sentry/node';
+import { QueryAuditLogsArgs } from '../../../__generated__/types.next';
+import { User } from '../../../shared/entities';
+import { ClickHouse, sql } from '../../operations/providers/clickhouse-client';
+import { SqlValue } from '../../operations/providers/sql';
+import { Logger } from '../../shared/providers/logger';
+import { formatToClickhouseDateTime } from '../helpers';
+import { AuditLogEvent, auditLogSchema } from './audit-logs-types';
+
+export const AUDIT_LOG_CLICKHOUSE_OBJECT = z.object({
+  id: z.string(),
+  event_time: z.string(),
+  user_id: z.string(),
+  user_email: z.string(),
+  organization_id: z.string(),
+  event_action: z.string(),
+  metadata: z.string().transform(x => JSON.parse(x)),
+});
+
+export type AuditLogModel = z.infer<typeof AUDIT_LOG_CLICKHOUSE_OBJECT>;
+
+const AUDIT_LOG_CLICKHOUSE_ARRAY = z.array(AUDIT_LOG_CLICKHOUSE_OBJECT);
```

Comment:
Please use "upper camel case" naming with "Model" as the ending for zod models. E.g. "AuditLogClickhouseArray"